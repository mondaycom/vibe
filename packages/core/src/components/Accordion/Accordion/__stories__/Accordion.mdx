import Accordion from "../Accordion";
import AccordionItem from "../../AccordionItem/AccordionItem";
import { Meta } from "@storybook/blocks";
import {
  BREADCRUBMS,
  EXPAND_COLLAPSE,
  TABLE
} from "../../../../storybook/components/related-components/component-description-map";
import { UsageGuidelines } from "vibe-storybook-components";
import * as AccordionStories from "./Accordion.stories";

<Meta of={AccordionStories} />

# Accordion

- [Overview](#overview)
- [Props](#props)
- [Usage](#usage)
- [Variants](#variants)
- [Do’s and don’ts](#dos-and-donts)
- [Use cases and examples](#use-cases-and-examples)
- [Related components](#related-components)
- [Feedback](#feedback)

## Overview

Accordion is a vertically stacked list of items. Each item can be "expanded" or "collapsed" to reveal the content within with that item.

<Canvas of={AccordionStories.Overview} />

### Import

```js
import { Accordion, AccordionItem } from "@vibe/core";
```

## Props

<PropsTable />

## Usage

<UsageGuidelines
  guidelines={[
    "Use accordion reduce clutter and chunk the information one by one",
    "Accordion Label must be short, clear, and understandable to indicate what's inside",
    "Default state of accordion is closed unless used for navigation",
    "Accordion content can include icons, radio buttons, and checkboxes"
  ]}
/>

## Variants

### Multi active

Each section can be expanded without closing the others

<Canvas of={AccordionStories.MultiActive} />

### Single active

Only one section can be open at the time

<Canvas of={AccordionStories.SingleActive} />

## Do’s and Don’ts

<ComponentRules
  rules={[
    {
      className: "monday-storybook-accordion_big-figure",
      positive: {
        component: (
          <Accordion className="monday-storybook-accordion_box-small">
            <AccordionItem title="Notifications" />
            <AccordionItem title="Security" />
            <AccordionItem title="InfoAssets" />
          </Accordion>
        ),
        description: "Use informative short labels."
      },
      negative: {
        className: "monday-storybook-accordion_big-figure",
        component: (
          <Accordion className="monday-storybook-accordion_box-small">
            <AccordionItem title="Closed" />
            <AccordionItem title="Closed" />
            <AccordionItem title="Closed" />
          </Accordion>
        ),
        description: "Indicate the accordion state with labels."
      }
    }
  ]}
/>

## Use cases and examples

### Preferences Accordion

<Canvas of={AccordionStories.PreferencesAccordion} />

## Related components

<RelatedComponents componentsNames={[EXPAND_COLLAPSE, TABLE, BREADCRUBMS]} />
