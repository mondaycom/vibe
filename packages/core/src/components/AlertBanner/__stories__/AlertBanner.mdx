import { Canvas, Meta } from "@storybook/blocks";
import { ComponentRules, Link, UsageGuidelines } from "vibe-storybook-components";
import AlertBanner from "../AlertBanner";
import AlertBannerText from "../AlertBannerText/AlertBannerText";
import AlertBannerLink from "../AlertBannerLink/AlertBannerLink";
import AlertBannerButton from "../AlertBannerButton/AlertBannerButton";
import {
  ATTENTION_BOX,
  TOAST,
  TOOLTIP
} from "../../../storybook/components/related-components/component-description-map";
import { TipWhenToUse } from "./AlertBanner.stories.helpers";
import * as AlertBannerStories from "./AlertBanner.stories";
import Flex from "../../Flex/Flex";

<Meta of={AlertBannerStories} />

# Alert Banner

- [Overview](#overview)
- [Props](#props)
- [Usage](#usage)
- [Variants](#variants)
- [Do’s and don’ts](#dos-and-donts)
- [Use cases and examples](#use-cases-and-examples)
- [Related components](#related-components)
- [Feedback](#feedback)

## Overview

Alert banners show pressing and high-signal messages, such as system alerts. They are meant to be noticed and prompt users to take action.

<Canvas of={AlertBannerStories.Overview} />

### Import

```js
import { AlertBanner } from "@vibe/core";
```

## Props

<PropsTable />

## Usage

<UsageGuidelines
  guidelines={[
    "Alert banners should have call for action and an option to dismiss.",
    "Don't include more than one call to action in an alert banner.",
    "Place the banner on the top of the screen, and push all other content below it."
  ]}
/>

<TipWhenToUse />

## Variants

### Types

There are five types of alert banners: primary, positive, negative, warning and inverted.

<Canvas of={AlertBannerStories.Types} />

### Alert Banner with button

<Canvas of={AlertBannerStories.AlertBannerWithButton} />

### Alert Banner with link

<Canvas of={AlertBannerStories.AlertBannerWithLink} />

## Do’s and Don’ts

<ComponentRules
  rules={[
    {
      positive: {
        component: (
          <div style={{ width: "392px" }}>
            <AlertBanner>
              <AlertBannerText text="Join us at Elevate!" />
              <AlertBannerLink text="RSVP now" href="https://monday.com" />
            </AlertBanner>
          </div>
        ),
        description: "Use banners for system messages, background processes, and general updates."
      },
      negative: {
        component: (
          <div style={{ width: "392px" }}>
            <AlertBanner>
              <AlertBannerText text="We successfully archived 1 item" />
            </AlertBanner>
          </div>
        ),
        description: (
          <>
            Don’t use banners for notifying a user of an action they have taken. Instead, provide visual feedback with a{" "}
            <StorybookLink page="Components/Toast">Toast.</StorybookLink>
          </>
        )
      }
    },
    {
      positive: {
        component: (
          <div style={{ width: "392px" }}>
            <AlertBanner backgroundColor="negative">
              <AlertBannerText text="No connection" />
              <AlertBannerLink text="Learn more" href="https://monday.com" />
              <AlertBannerButton>Try again</AlertBannerButton>
            </AlertBanner>
          </div>
        ),
        description: "If two actions are needed, use two different call to acitons."
      },
      negative: {
        component: (
          <Flex direction="column" gap={16} style={{ width: "392px" }}>
            <AlertBanner backgroundColor="negative">
              <AlertBannerText text="No connection" />
              <AlertBannerLink text="Learn more" href="https://monday.com" />
              <AlertBannerLink text="Try again" href="https://monday.com" />
            </AlertBanner>
            <AlertBanner backgroundColor="negative">
              <AlertBannerText text="No connection" />
              <AlertBannerButton>Learn more</AlertBannerButton>
              <AlertBannerButton>Try again</AlertBannerButton>
            </AlertBanner>
          </Flex>
        ),
        description: "Don't include more than one action in an alert banner with the same type."
      }
    },
    {
      positive: {
        component: (
          <div style={{ width: "392px" }}>
            <AlertBanner backgroundColor="dark">
              <AlertBannerText text="Join us at Elevate!" />
              <AlertBannerLink text="RSVP now" href="https://monday.com" />
            </AlertBanner>
          </div>
        ),
        description: "Use only the 4 color types: primary, negative, positive, and inverted."
      },
      negative: {
        component: (
          <div style={{ width: "392px" }}>
            <AlertBanner>
              <AlertBannerText text="Join us at Elevate!" />
              <AlertBannerLink text="RSVP now" href="https://monday.com" />
            </AlertBanner>
          </div>
        ),
        description: "Don’t choose other colors for alert banners. Keep it consistent."
      }
    }
  ]}
/>

## Use cases and examples

### Alert banner as an announcement

Use when you'd like to notify about an event or cross-company announcment.

<Canvas of={AlertBannerStories.AlertBannerAsAnAnnouncement} />

### Alert banner as an opportunity to upgrade

Use to show a trial user the number of remaining free days to use the platform.

<Canvas of={AlertBannerStories.AlertBannerAsAnOpportunityToUpgrade} />

### Overflow text

In case that there's not enough space for the content, use an ellipses (...).

<Canvas of={AlertBannerStories.OverflowText} />

## Related components

<RelatedComponents componentsNames={[TOAST, ATTENTION_BOX, TOOLTIP]} />
