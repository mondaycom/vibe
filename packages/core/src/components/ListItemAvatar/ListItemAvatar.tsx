import React, { forwardRef, useRef } from "react";
import cx from "classnames";
import useMergeRef from "../../hooks/useMergeRef";
import { VibeComponentProps, withStaticProps } from "../../types";
import Avatar from "../Avatar/Avatar";
import { ListItemComponentType as ListItemComponentTypeEnum } from "../ListItem/ListItemConstants";
import { ListItemElement } from "../ListItem";
import styles from "./ListItemAvatar.module.scss";

export interface ListItemAvatarProps extends VibeComponentProps {
  /**
   * The HTML element used for the list item avatar.
   */
  component?: ListItemElement;
  /**
   * The source URL of the avatar image.
   */
  src?: string;
  /**
   * Class name applied to the avatar.
   */
  avatarClassName?: string;
}

const ListItemAvatar = forwardRef(
  (
    { className, id, src, avatarClassName, component: Component = "div" }: ListItemAvatarProps,
    ref: React.ForwardedRef<HTMLElement>
  ) => {
    const componentRef = useRef(null);
    const mergedRef = useMergeRef(ref, componentRef);

    return (
      <Component ref={mergedRef} className={cx(styles.listItemAvatar, className)} id={id} aria-hidden="true">
        <Avatar src={src} type="img" size="small" className={cx(styles.avatar, avatarClassName)} />
      </Component>
    );
  }
);

interface ListItemAvatarStaticProps {
  components: typeof ListItemComponentTypeEnum;
}

export default withStaticProps<ListItemAvatarProps, ListItemAvatarStaticProps>(ListItemAvatar, {
  components: ListItemComponentTypeEnum
});
