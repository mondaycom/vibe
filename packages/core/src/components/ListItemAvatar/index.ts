export { default as ListItemAvatar, type ListItemAvatarProps } from "./ListItemAvatar";
