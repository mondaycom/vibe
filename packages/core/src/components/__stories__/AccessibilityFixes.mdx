import { Meta, Canvas } from "@storybook/blocks";
import * as AccessibilityFixesStories from "./AccessibilityFixes.stories";

<Meta of={AccessibilityFixesStories} />

# Accessibility Fixes

## TextField

Removed the hidden buttons from the dom when no icon is shown

<Canvas of={AccessibilityFixesStories.TextFieldConditionalIcons} />

## Checkbox

The label is now associated with the input using the id prop, and does not wrap the input.

<Canvas of={AccessibilityFixesStories.CheckboxJAWSCompatibility} />

## MenuButton

Remove the role="dialog" as the list role is sufficient

<Canvas of={AccessibilityFixesStories.MenuButtonAccessibility} />

## TextArea

When there is a character limit aria-describedby is used to describe the character limit: "44 out of 2000"

<Canvas of={AccessibilityFixesStories.TextAreaCharacterLimit} />
