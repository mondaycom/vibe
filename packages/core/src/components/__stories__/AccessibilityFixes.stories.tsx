import React, { useState } from "react";
import { Meta, StoryObj } from "@storybook/react";
import TextField from "../TextField/TextField";
import Checkbox from "../Checkbox/Checkbox";
import MenuButton from "../MenuButton/MenuButton";
import TextArea from "../TextArea/TextArea";
import Menu from "../Menu/Menu/Menu";
import MenuItem from "../Menu/MenuItem/MenuItem";
import { createStoryMetaSettingsDecorator } from "../../storybook";
import { CloseSmall, Sun, Moon, Favorite } from "@vibe/icons";

type Story = StoryObj;

const metaSettings = createStoryMetaSettingsDecorator({
  component: TextField
});

export default {
  title: "Internal/Accessibility Fixes",
  decorators: metaSettings.decorators
} satisfies Meta;

const TextFieldConditionalExample = () => {
  const [searchValue, setSearchValue] = useState("");

  return (
    <div style={{ width: "400px" }}>
      <TextField
        title="Text Field"
        placeholder="Placeholder text"
        value={searchValue}
        onChange={value => setSearchValue(value)}
        iconName={searchValue ? CloseSmall : undefined}
        iconsNames={{ primary: searchValue ? "Close" : undefined, secondary: searchValue ? "Close" : undefined }}
        clearOnIconClick
      />
    </div>
  );
};

export const TextFieldConditionalIcons: Story = {
  render: TextFieldConditionalExample,
  name: "TextField - Conditional Icons"
};

export const CheckboxJAWSCompatibility: Story = {
  render: () => (
    <div style={{ width: "400px" }}>
      <Checkbox id="terms-checkbox" label="I agree to the terms and conditions" defaultChecked={true} />
    </div>
  ),
  name: "Checkbox - JAWS Compatibility"
};

const MenuButtonExample = () => {
  return (
    <div style={{ width: "400px" }}>
      <MenuButton>
        <Menu id="accessibility-menu" size="medium" focusItemIndexOnMount={0}>
          <MenuItem icon={Sun} iconType="svg" title="Light theme" />
          <MenuItem icon={Moon} iconType="svg" title="Dark theme" />
          <MenuItem icon={Favorite} iconType="svg" title="Favorites" />
        </Menu>
      </MenuButton>
    </div>
  );
};

export const MenuButtonAccessibility: Story = {
  render: MenuButtonExample,
  name: "MenuButton - Accessibility"
};

const TextAreaExample = () => {
  const [textValue, setTextValue] = useState("");

  return (
    <div style={{ width: "400px" }}>
      <TextArea
        label="Message"
        placeholder="Type your message..."
        value={textValue}
        onChange={e => setTextValue(e.target.value)}
        maxLength={100}
        showCharCount
        allowExceedingMaxLength
      />
    </div>
  );
};

export const TextAreaCharacterLimit: Story = {
  render: TextAreaExample,
  name: "TextArea - Character Limit"
};
