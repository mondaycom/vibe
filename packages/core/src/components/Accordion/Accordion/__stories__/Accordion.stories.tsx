import React from "react";
import Accordion, { AccordionProps } from "../Accordion";
import AccordionItem from "../../AccordionItem/AccordionItem";
import Checkbox from "../../../Checkbox/Checkbox";
import { accordionMultiActivePlaySuite, accordionSingleActivePlaySuite } from "../__tests__/Accordion.interactions";
import "./Accordion.stories.scss";

const accordionTemplate = (args: AccordionProps) => {
  return (
    <Accordion className="monday-storybook-accordion_small-wrapepr" defaultIndex={[1]} {...args}>
      <AccordionItem title="Notifications">
        <div className="monday-storybook-accordion_small-box" />
      </AccordionItem>
      <AccordionItem title="Setting">
        <div className="monday-storybook-accordion_small-box" />
      </AccordionItem>
      <AccordionItem title="Info">
        <div className="monday-storybook-accordion_small-box" />
      </AccordionItem>
      <AccordionItem title="Profile">
        <div className="monday-storybook-accordion_small-box" />
      </AccordionItem>
      <AccordionItem title="Permissions">
        <div className="monday-storybook-accordion_small-box" />
      </AccordionItem>
      <AccordionItem title="Security">
        <div className="monday-storybook-accordion_small-box" />
      </AccordionItem>
      <AccordionItem title="Connectivity">
        <div className="monday-storybook-accordion_small-box" />
      </AccordionItem>
      <AccordionItem title="Integration">
        <div className="monday-storybook-accordion_small-box" />
      </AccordionItem>
      <AccordionItem title="Assets">
        <div className="monday-storybook-accordion_small-box" />
      </AccordionItem>
    </Accordion>
  );
};

export default {
  title: "Components/Accordion",
  component: Accordion,

  subcomponents: {
    AccordionItem
  },

  argTypes: {
    children: {
      control: false
    },

    defaultIndex: {
      control: false
    }
  }
};

export const Overview = {
  render: accordionTemplate.bind({}),
  name: "Overview",
  args: {},
  parameters: {
    docs: {
      liveEdit: {
        isEnabled: false
      }
    }
  }
};

export const MultiActive = {
  render: () => (
    <Accordion className="monday-storybook-accordion_small-wrapepr" allowMultiple defaultIndex={[1, 3]}>
      <AccordionItem title="Notifications">
        <div className="monday-storybook-accordion_small-box" />
      </AccordionItem>
      <AccordionItem title="Setting">
        <div className="monday-storybook-accordion_small-box" />
      </AccordionItem>
      <AccordionItem title="Connectivity">
        <div className="monday-storybook-accordion_small-box" />
      </AccordionItem>
      <AccordionItem title="Integration">
        <div className="monday-storybook-accordion_small-box" />
      </AccordionItem>
      <AccordionItem title="Assets">
        <div className="monday-storybook-accordion_small-box" />
      </AccordionItem>
    </Accordion>
  ),

  name: "Multi active",
  play: accordionMultiActivePlaySuite
};

export const SingleActive = {
  render: () => (
    <Accordion className="monday-storybook-accordion_small-wrapepr" defaultIndex={[1]}>
      <AccordionItem title="Notifications">
        <div className="monday-storybook-accordion_small-box" />
      </AccordionItem>
      <AccordionItem title="Setting">
        <div className="monday-storybook-accordion_small-box" />
      </AccordionItem>
      <AccordionItem title="Connectivity">
        <div className="monday-storybook-accordion_small-box" />
      </AccordionItem>
      <AccordionItem title="Integration">
        <div className="monday-storybook-accordion_small-box" />
      </AccordionItem>
      <AccordionItem title="Assets">
        <div className="monday-storybook-accordion_small-box" />
      </AccordionItem>
    </Accordion>
  ),

  name: "Single active",
  play: accordionSingleActivePlaySuite
};

export const PreferencesAccordion = {
  render: () => (
    <Accordion className="monday-storybook-accordion_wrapper" defaultIndex={[0]}>
      <AccordionItem title="In monday">
        <div className="monday-storybook-accordion_checkbox-wrapper">
          <Checkbox label="Likes my update" checked />
          <Checkbox label="Replies to my update" />
          <Checkbox label="Replies or likes a conversation I'm a part of" checked />
          <Checkbox label="Subscribes me to a Board/Item/Team" checked />
          <Checkbox label="Writes an update on an items I'm subscribed to" checked />
        </div>
      </AccordionItem>
    </Accordion>
  ),

  name: "Preferences Accordion"
};
