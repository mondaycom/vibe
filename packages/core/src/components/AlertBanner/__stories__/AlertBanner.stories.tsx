import React from "react";
import AlertBanner, { AlertBannerProps } from "../AlertBanner";
import AlertBannerText from "../AlertBannerText/AlertBannerText";
import AlertBannerLink from "../AlertBannerLink/AlertBannerLink";
import AlertBannerButton from "../AlertBannerButton/AlertBannerButton";
import { createStoryMetaSettingsDecorator } from "../../../storybook/functions/createStoryMetaSettingsDecorator";
import "./AlertBanner.stories.scss";
import { Meta, StoryObj } from "@storybook/react";

const metaSettings = createStoryMetaSettingsDecorator({
  component: AlertBanner,
  actionPropsArray: ["onClose"]
});

const alertBannerTemplate = (args: AlertBannerProps) => {
  return (
    <AlertBanner {...args}>
      <AlertBannerText text="Alert banner message" />
      <AlertBannerLink text="this is a CTA" href="https://monday.com" />
    </AlertBanner>
  );
};

export default {
  title: "Components/AlertBanner",
  component: AlertBanner,
  subcomponents: {
    AlertBannerText,
    AlertBannerLink,
    AlertBannerButton
  },
  argTypes: metaSettings.argTypes,
  decorators: metaSettings.decorators
} satisfies Meta<typeof AlertBanner>;

export const Overview = {
  render: alertBannerTemplate.bind({}),
  name: "Overview",

  args: {
    className: "monday-storybook-alert-banner_big-container"
  }
};

type Story = StoryObj<typeof AlertBanner>;

export const Types: Story = {
  render: () => (
    <div className="monday-storybook-alert-banner_column-wrapper monday-storybook-alert-banner_big-container">
      <AlertBanner>
        <AlertBannerText text="Alert banner message" />
        <AlertBannerLink text="this is a CTA" href="https://monday.com" />
      </AlertBanner>
      <AlertBanner backgroundColor="positive">
        <AlertBannerText text="Alert banner message" />
        <AlertBannerLink text="this is a CTA" href="https://monday.com" />
      </AlertBanner>
      <AlertBanner backgroundColor="negative">
        <AlertBannerText text="Alert banner message" />
        <AlertBannerLink text="this is a CTA" href="https://monday.com" />
      </AlertBanner>
      <AlertBanner backgroundColor="warning">
        <AlertBannerText text="Alert banner message" />
        <AlertBannerLink text="this is a CTA" href="https://monday.com" />
      </AlertBanner>
      <AlertBanner backgroundColor="dark">
        <AlertBannerText text="Alert banner message" />
        <AlertBannerLink text="this is a CTA" href="https://monday.com" />
      </AlertBanner>
    </div>
  ),

  name: "Types"
};

export const AlertBannerWithButton: Story = {
  render: () => (
    <AlertBanner className="monday-storybook-alert-banner_big-container">
      <AlertBannerText text="Lorem ipsum dolor sit amet" />
      <AlertBannerButton>Lorem Ipsum</AlertBannerButton>
    </AlertBanner>
  ),

  name: "Alert Banner with button"
};

export const AlertBannerWithLink: Story = {
  render: () => (
    <AlertBanner className="monday-storybook-alert-banner_big-container">
      <AlertBannerText text="Alert banner message" />
      <AlertBannerLink text="this is a CTA" href="https://monday.com" />
    </AlertBanner>
  ),

  name: "Alert Banner with link"
};

export const AlertBannerAsAnAnnouncement: Story = {
  render: () => (
    <AlertBanner backgroundColor="dark" className="monday-storybook-alert-banner_big-container">
      <AlertBannerText text="Join us at Elevate 2022" />
      <AlertBannerLink text="RSVP now" href="https://monday.com" />
    </AlertBanner>
  ),

  name: "Alert banner as an announcement"
};

export const AlertBannerAsAnOpportunityToUpgrade: Story = {
  render: () => (
    <AlertBanner className="monday-storybook-alert-banner_big-container">
      <AlertBannerText text="7 days left on your monday CRM trial" />
      <AlertBannerLink text="Upgrade now" href="https://monday.com" />
    </AlertBanner>
  ),

  name: "Alert banner as an opportunity to upgrade"
};

export const OverflowText: Story = {
  render: () => (
    <AlertBanner className="monday-storybook-alert-banner_small-container">
      <AlertBannerText text="This is a really long alert..." />
      <AlertBannerLink text="Call to action" href="https://monday.com" />
    </AlertBanner>
  ),

  name: "Overflow text"
};
