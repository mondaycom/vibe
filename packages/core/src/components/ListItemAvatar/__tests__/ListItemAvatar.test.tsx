import { it, expect } from "vitest";
import React from "react";
import renderer from "react-test-renderer";
import ListItemAvatar from "../ListItemAvatar";
import { person1 } from "../../Avatar/__stories__/assets";

it("renders correctly with empty props", () => {
  const tree = renderer.create(<ListItemAvatar />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly with picture", () => {
  const tree = renderer.create(<ListItemAvatar src={person1} />).toJSON();
  expect(tree).toMatchSnapshot();
});
