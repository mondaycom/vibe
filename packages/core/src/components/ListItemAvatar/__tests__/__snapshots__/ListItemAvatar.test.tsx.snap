// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly with empty props 1`] = `
<div
  aria-hidden="true"
  className="listItemAvatar"
>
  <div
    className="avatar small avatar"
    data-testid="avatar"
    style={{}}
  >
    <div
      aria-hidden={false}
      className="circle circleImg"
      style={
        {
          "backgroundColor": "var(--color-chili-blue)",
        }
      }
      tabIndex={-1}
    >
      <img
        className="contentImg contentImgSmall"
        data-testid="avatar-content"
      />
    </div>
  </div>
</div>
`;

exports[`renders correctly with picture 1`] = `
<div
  aria-hidden="true"
  className="listItemAvatar"
>
  <div
    className="avatar small avatar"
    data-testid="avatar"
    style={{}}
  >
    <div
      aria-hidden={false}
      className="circle circleImg"
      style={{}}
      tabIndex={-1}
    >
      <img
        className="contentImg contentImgSmall"
        data-testid="avatar-content"
        src="/src/components/Avatar/__stories__/assets/person1.png"
      />
    </div>
  </div>
</div>
`;
